extras/ioc_hash_scan.sh

```diff
@@ -0,0 +1,189 @@
+  serverList+=("$line")
+done < "$1"
+
+if [ ${#serverList[@]} -eq 0 ]; then
+  echo No Instances found in "$1" - exiting.
+  exit 1
+fi
+
+# Path of aws credentials file
+AWSPROF="$2"
+
+today=$(date +%Y-%m-%d)
+logfile="$HOME/$today-ioc-scanner-hashscan.log"
+
+# Suppress some verbose stdout.
+# Suppress stnderr of pkill command
+exec > >(grep --invert-match 'Starting\|Exiting')
+exec 2> >(grep --invert-match 'SIGTERM')
+
+## FUNCTIONS
+function getOSType() {
+  OSNAME=$(AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+    aws --profile="$AWSPROF" --region="$AWS_REGION" \
+    ssm start-session --target="$i" \
+    --document=AWS-StartInteractiveCommand \
+    --parameters="command=cat /etc/os-release | grep --extended-regexp ^NAME=" \
+    --output text)
+  echo "$OSNAME" | grep NAME | cut -d'"' -f2 | cut -d' ' -f1
+}
+
+function getHost() {
+  HOST=$(AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+    aws --profile="$AWSPROF" --region="$AWS_REGION" \
+    ssm start-session --target="$i" \
+    --document=AWS-StartInteractiveCommand \
+    --parameters="command=hostname" \
+    --output text)
+  echo "$HOST" | grep --invert-match "session" | sed '/^$/d'
+}
+
+function portForward() {
+  AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+    aws --profile="$AWSPROF" --region="$AWS_REGION" \
+    ssm start-session --target="$i" \
+    --document=AWS-StartPortForwardingSession --parameters="localPortNumber=5555,portNumber=6666"
+}
+
+function installNC() {
+  if [[ "$OS" == "Debian" ]]; then
+    AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+      aws --profile="$AWSPROF" --region="$AWS_REGION" \
+      ssm start-session --target="$i" \
+      --document=AWS-StartInteractiveCommand \
+      --parameters="command='sudo apt-get install netcat -y'"
+  else
+    AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+      aws --profile="$AWSPROF" --region="$AWS_REGION" \
+      ssm start-session --target="$i" \
+      --document=AWS-StartInteractiveCommand \
+      --parameters="command='sudo yum install netcat -y'"
+  fi
+}
+
+function startListen() {
+  if [[ "$OS" == "Debian" ]]; then
+    AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+      aws --profile="$AWSPROF" --region="$AWS_REGION" \
+      ssm start-session --target="$i" \
+      --document=AWS-StartInteractiveCommand \
+      --parameters="command='cd ~/src/ioc_scan; nc -l -p 6666 | tar xzf -'"
+  else
+    AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+      aws --profile="$AWSPROF" --region="$AWS_REGION" \
+      ssm start-session --target="$i" \
+      --document=AWS-StartInteractiveCommand \
+      --parameters="command='cd ~/src/ioc_scan; nc -l 6666 | tar xzf -'"
+  fi
+}
+
+## MAIN SCRIPT
+echo "IOC Hash Scan - $today-$(date +%H:%M:%S)" > "$logfile"
+
+for i in "${serverList[@]}"; do
+  OS="$(getOSType)"
+  if [[ "$OS" != "Debian" && "$OS" != "Fedora" ]]; then
+    echo "Non-supported OS Type"
```

Comment:
Improve error message:
```suggestion
    echo "Instance $i is running an operating system ($OS) that is not supported by this script."
    echo "Currently only Debian and Fedora are supported - exiting."
```